mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -502,4 +504,55 @@ def GENX_MatrixCopyOp : GENX_Op<"matrix.copy">,
   let hasVerifier = 1;
 }
 
+def GENX_YieldOp : GENX_Op<"yield", [Pure, ReturnLike, Terminator]>,
+    Arguments<(ins Variadic<AnyType>:$values)> {
+  let summary = "Terminates and yields values from GENX regions";
+
+  string baseDescription = [{
+    The `genx.yield` operation yields one or more SSA value from a GENX dialect op 
+    region and terminates the region. The operands of this operation must correspond 
+    to the parent operation results.
+  }];
+
+  let builders = [OpBuilder<(ins), [{ /* nothing to do */ }]>];
+  let assemblyFormat = "attr-dict ($values^ `:` type($values))?";
+  let hasVerifier = 1;
+}
+
+def GENX_MatrixMapOp : GENX_Op<"matrix.map", [
+        DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+        RecursiveMemoryEffects,
+        SingleBlockImplicitTerminator<"YieldOp">,
+        TypesMatchWith<"result #0 and operand #0 have the same type",
+                       "res", "mat", "$_self">]>,
+    Results<(outs GENX_AnyJointMatrix:$res)>,    
+    Arguments<(ins GENX_AnyJointMatrix:$mat, Variadic<IntegerOrFloatType>:$inputs, ScopeAttr:$scope)> {
+  let summary = "Elementwise operation on a joint matrix";
+
+  string baseDescription = [{
+    The `genx.matrix.map` operation performs an element-wise on the joint matrix $mat.
+    The region attached to the operation should contain the workload and it is applied 
+    to each element of the matrix.
+
+    Example:
+
+    %res = genx.matrix.map <Subgroup>
+    ins(%mat, %val : !genx.jointmatrix<8x32xf32, RowMajor>, f32)
+    (%elem: f32, %v: f32) {
+       %0 = arith.addf %elem, %v : f32
+       genx.yield %0 : f32
+    } : !genx.jointmatrix<8x32xf32, RowMajor>
+  }];
+
+  let builders = [
+    OpBuilder<(ins "Value":$mat, "ValueRange":$inputs, 
```

Comment:
Can we have a target lit test to test this portion of the code?

Reply:
To exercise the builder we would need to create the operation (e.g. rewriter.create<genx::MapOp>(...)) so that cannot be done using lit tests.

Reply:
Ok I added one with an assert. Codegen to be done in a separate PR.

Reply:
Thanks, should we add one for yield as well?